### PR TITLE
Consolidate evaluation context merge order in the spec

### PR DESCRIPTION
**Pull request by beeme1mr**

### CODE REVIEW

**Comment by beeme1mr** [comment]
FYI, @rgrassian-split @patricioe

**Review by skyerus — APPROVED** [review]
Seems clear to me, nice work

**Review by benjiro — APPROVED** [review]

File: specification/sections/03-evaluation-context.md [hunk 1]
```diff
@@ -52,6 +52,17 @@ API (global) `evaluation context` can be used to supply static data to flag eval
 
 #### Requirement 3.2.2
 
-> Evaluation context **MUST** be merged in the order: API (global) -> client -> invocation, with duplicate values being overwritten.
+> Evaluation context **MUST** be merged in the order: API (global) -> client -> invocation -> before hooks, with duplicate values being overwritten.
```
Comment (toddbaert) [inline specification/sections/03-evaluation-context.md:55]: I liked @justinabrahms' language that included (hightest priority) and (lowest priority). It was very clear to me:

>before-hook (highest precedence), invocation, client, api (lowest precedence)

File: specification/sections/03-evaluation-context.md [hunk 2]
```diff
@@ -52,6 +52,17 @@ API (global) `evaluation context` can be used to supply static data to flag eval
...
+```mermaid
+flowchart LR
+  global("API (global)")
+  client("Client")
+  invocation("Invocation")
+  hook("Before Hooks")
+  global --> client
+  client --> invocation
+  invocation --> hook
+```
```
Comment (toddbaert) [inline specification/sections/03-evaluation-context.md:68]: This is great. The mermaid was a good idea.

**Review by toddbaert — APPROVED** [review]

File: specification/sections/03-evaluation-context.md (hunk 1 above)
Comment (justinabrahms) [reply specification/sections/03-evaluation-context.md:55]:
```suggestion
> Evaluation context **MUST** be merged in the order: API (global; lowest precedence) -> client -> invocation -> before hooks (highest precedence), with duplicate values being overwritten.
```

**Review by justinabrahms — APPROVED** [review]

**Pull request merged**